.github/actions/build-helm/action.yml

```diff
@@ -0,0 +1,165 @@
+name: 'Build, test, package and push helm chart'
+description: 'Build, test, package and maybe push helm chart'
+inputs:
+  chart-dir:
+    description: 'Path to the folder holding Chart.yaml'
+    required: true
+  helm-repository:
+    description: 'Charts repository to checkout'
+    required: true
+  helm-repository-branch:
+    description: 'Branch on the charts repository'
+    required: false
+  helm-repository-subfolder:
+    description: 'Subfolder on the charts repository to receive changes'
+    required: false
+  helm-repository-token:
+    description: 'The Github token to checkout the charts repository'
+    required: true
+  helm-repository-commit-username:
+    description: 'The username to commit on the charts repository'
+    required: true
+  helm-repository-url:
+    description: 'Charts repository URL'
+    required: true
+  test-rancher-url:
+    description: 'Rancher URL for tests, tests are skipped if not filled'
+    required: false
+  test-rancher-access-key:
+    description: 'Rancher access key for tests, tests are skipped if not filled'
+    required: false
+  test-rancher-secret-key:
+    description: 'Rancher secret key for tests, tests are skipped if not filled'
+    required: false
+  test-cluster-name:
+    description: 'Name of the Rancher cluster name for tests'
+    required: false
+  test-namespace:
+    description: >
+      Name of the namespace in rancher for tests (should be unique in PRs),
+      tests are skipped if not filled.
+    required: false
+  do-push:
+    description: "If 'true', changes will be pushed ('false' can be used for PRs)"
+    default: 'false'
+    required: false
+
+runs:
+  using: composite
+  steps:
+
+    - name: Build
+      uses: Alfresco/alfresco-build-tools/.github/actions/build-helm-chart@v1.9.2
+      with:
+        chart-dir: ${{ inputs.chart-dir }}
+
+    - name: Test
+      env:
+        CHART_DIR: ${{ inputs.chart-dir }}
+      shell: bash
+      run: |
+        helm plugin install https://github.com/quintush/helm-unittest
+        helm unittest $CHART_DIR --helm3
+
+    - name: Check if can dry run helm chart
+      id: check
+      env:
+        NAMESPACE: ${{ inputs.test-namespace }}
+        CLUSTER: ${{ inputs.test-cluster-name }}
+        URL: ${{ inputs.test-rancher-url }}
+        ACCESS_KEY: ${{ inputs.test-rancher-access-key }}
+        SECRET_KEY: ${{ inputs.test-rancher-secret-key }}
```

Comment:
I would treat testing on a live environment as a separate action that you call or not in a specific repo depending if you want to have that or not (in a near future we may want to try KinD that could supersede testing on clusters managed by rancher)